src/components/PatchActionButtons/RestartPatch.tsx

```diff
@@ -9,47 +9,56 @@ interface RestartPatchProps {
   disabled: boolean;
   isButton?: boolean;
   refetchQueries: string[];
+  visibilityControl?: [boolean, React.Dispatch<React.SetStateAction<boolean>>];
   hideMenu: (e?: React.MouseEvent<HTMLElement, MouseEvent>) => void;
 }
-export const RestartPatch = forwardRef<HTMLDivElement, RestartPatchProps>(
-  ({ isButton, disabled, patchId, refetchQueries, hideMenu }, ref) => {
-    const [openModal, setOpenModal] = useState(false);
-    return (
-      <>
-        {isButton ? (
-          <Button
-            size="small"
-            dataCy="restart-patch"
-            disabled={disabled}
-            loading={false}
-            onClick={() => setOpenModal(!openModal)}
-          >
-            Restart
-          </Button>
-        ) : (
-          <DropdownItem
-            disabled={disabled}
-            data-cy="restart-patch"
-            onClick={() => setOpenModal(!openModal)}
-          >
-            <Disclaimer>Restart</Disclaimer>
-          </DropdownItem>
-        )}
-        {openModal && (
-          <div className={openModal ? "ant-popover-open" : ""} ref={ref}>
-            <PatchRestartModal
-              patchId={patchId}
-              visible={openModal}
-              onOk={() => {
-                setOpenModal(false);
-                hideMenu();
-              }}
-              onCancel={() => setOpenModal(false)}
-              refetchQueries={refetchQueries}
-            />
-          </div>
-        )}
-      </>
-    );
-  }
-);
+export const RestartPatch: React.FC<RestartPatchProps> = ({
+  isButton,
+  disabled,
+  patchId,
+  refetchQueries,
+  hideMenu,
+  visibilityControl,
+}) => {
+  const fallbackVisibilityControl = useState(false);
+  const [isVisible, setIsVisible] =
+    visibilityControl !== undefined
+      ? visibilityControl
+      : fallbackVisibilityControl;
```

Comment:
I'm not totally convinced we should have two methods of controlling the state of this component. What would be the advantage of doing it this way vs only one method?

Reply:
Ideally, the RestartPatch should be the only component to control the visibility of the modal because that way we do not need to declare a useState to manage to visibility whenever we import RestartPatch (we are able to avoid that here because the prop is optional: https://github.com/evergreen-ci/spruce/blob/master/src/pages/patch/ActionButtons.tsx#L89).